src/Kernels.cpp

```diff
@@ -3,132 +3,136 @@
 #include "Kernels.h"
 #include "Error.h"
 #include "Dist.h"
-#include "Param.h"
-
-// To speed up calculation of kernel values we provide a couple of lookup
-// tables.
-//
-// nKernel is a P.NKR+1 element table of lookups nKernel[0] is the kernel
-// value at a distance of 0, and nKernel[P.NKR] is the kernel value at the
-// largest possible distance (diagonal across the bounding box).
-//
-// nKernelHR is a higher-resolution table of lookups, also of P.NKR+1
-// elements.  nKernelHR[n * P.NK_HR] corresponds to nKernel[n] for
-// n in [0, P.NKR/P.NK_HR]
-//
-// Graphically:
-//
-// Distance 0            ...                              Bound Box diagonal
-//          nKernel[0]   ... nKernel[P.NKR / P.NK_HR] ... nKernel[P.NKR]
-//          nKernelHR[0] ... nKernelHR[P.NKR]
-double *nKernel, *nKernelHR;
-
-void InitKernel(double norm)
+
+using namespace CovidSim::TBD1;
+
+bool KernelLookup::setup(double longest_distance)
 {
-	double(*Kernel)(double);
-
-	if (P.KernelType == 1)
-		Kernel = ExpKernel;
-	else if (P.KernelType == 2)
-		Kernel = PowerKernel;
-	else if (P.KernelType == 3)
-		Kernel = GaussianKernel;
-	else if (P.KernelType == 4)
-		Kernel = StepKernel;
-	else if (P.KernelType == 5)
-		Kernel = PowerKernelB;
-	else if (P.KernelType == 6)
-		Kernel = PowerKernelUS;
-	else if (P.KernelType == 7)
-		Kernel = PowerExpKernel;
+	try
```

Comment:
I think this is a case of tell don't ask.

Let the function throw an exception instead of returning a boolean.

Reply:
Did think about that at the time but I'm always loath to change the behaviour of unfamiliar code so I thought I'd do the ugly solution. I'll change it and not do any check on return. Thanks.